packages/p2p/src/components/my-profile/payment-methods/add-payment-method/cancel-add-payment-method-modal.jsx

```diff
@@ -1,42 +1,12 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { Button, Modal, Text } from '@deriv/components';
 import { observer } from 'mobx-react-lite';
 import { useStores } from 'Stores';
 import { Localize } from 'Components/i18next';
-import { reaction } from 'mobx';
+import { isMobile } from '@deriv/shared';
 
-const CancelAddPaymentMethodModal = ({ is_floating }) => {
-    const { my_profile_store, my_ads_store } = useStores();
-
-    // TODO: Refactor this code to avoid manual DOM updates
-    // mounts the modal in a seperate modal-root container to show/float the modal over another modal if is_floating is true
-    React.useLayoutEffect(() => {
-        const disposeFloatingWrapper = reaction(
-            () => my_profile_store.is_cancel_add_payment_method_modal_open,
-            is_open => {
-                let wrapper = document.getElementById('cancel_modal_root');
-                if (is_open) {
-                    if (!wrapper) {
-                        wrapper = document.createElement('div');
-                        wrapper.setAttribute('id', 'cancel_modal_root');
-                    }
-                    if (is_floating) {
-                        wrapper.classList.add('modal-root');
-                        document.body.appendChild(wrapper);
-                    }
-                } else if (wrapper) {
-                    document.body.removeChild(wrapper);
-                }
-            }
-        );
-
-        return () => {
-            disposeFloatingWrapper();
-            my_profile_store.setSelectedPaymentMethod('');
-            my_profile_store.setSelectedPaymentMethodDisplayName('');
-        };
-    }, []);
+const CancelAddPaymentMethodModal = ({ onCancel = () => {}, onGoBack = () => {} }) => {
```

Comment:
why instantiate these as empty functions? also do we not need to pass it into propTypes as well?

Reply:
In some pages they do not need to pass in their own custom navigation handlers, in this case when `onCancel` and `onGoBack` is not needed, this is instantiated with an empty function as a default

Reply:
```suggestion
const CancelAddPaymentMethodModal = ({ onCancel, onGoBack }) => {
```
we can check if `onCancel` and `onGoBack` are defined before we call them below like so 🙂:
```onCancel?.();```
```if (onCancel) setTimeout(onCancel, 250);```
```onGoBack?.();```

Reply:
Oh yaa I should have done that, it looks better 😓